src/librustdoc/html/markdown.rs

```diff
@@ -420,29 +421,49 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for SummaryLine<'a, I> {
         if !self.started {
             self.started = true;
         }
-        let event = self.inner.next();
-        let mut is_start = true;
-        let is_allowed_tag = match event {
-            Some(Event::Start(ref c)) => {
-                self.depth += 1;
-                check_if_allowed_tag(c)
-            }
-            Some(Event::End(ref c)) => {
-                self.depth -= 1;
-                is_start = false;
-                check_if_allowed_tag(c)
+        while let Some(event) = self.inner.next() {
+            let mut is_start = true;
+            let is_allowed_tag = match event {
+                Event::Start(Tag::CodeBlock(_)) => {
```

Comment:
Why was the extra handling of the CodeBlock tag necessary?

Reply:
This is the whole point of the PR. :)

Reply:
I think i see it now. I had thought that just taking it out of `check_if_allowed_tag` above would work, but i guess that would then splice the entire code block as a code block instead of adding the code as a `<p>` tag. With these extra checks, we can properly skip it.

It does seem like you're making it pick the first non-code-block paragraph, though. Is that intended? Can you add a test to demonstrate this?

Reply:
I confirm it takes the next item. Do you want to keep this behavior or do you want me to just remove the short docblock in case it starts with a codeblock?

Reply:
I think i'd prefer if it removed the short docblock in that case.